File: proxy/tests/script_engine/req_set_method.py
from roxy import Extension, Method


class ChangeMethod(Extension):
    def request(self, flow):
        print(flow.request.method)
        if flow.request.method == Method.GET:
            print("setting")
            flow.request.method = Method.POST
        else:
            print("not settings")

        print(flow.request.method)


Extensions = [ChangeMethod()]

File: proxy/tests/script_engine/req_set_method.js
/// <reference path="../../script_libs/js/index.d.ts" />
/** @type {Extension} */
const reqSetMethod = {
  request(flow) {
    if (flow.request.method == Method.GET) {
      flow.request.method = Method.POST;
    }
  }
};
globalThis.extensions = [reqSetMethod];
